primitives/messages/src/source_chain.rs

```diff
@@ -135,6 +135,15 @@ pub trait MessageDeliveryAndDispatchPayment<AccountId, Balance> {
 	}
 }
 
+/// Handler for delivered messages.
+#[impl_trait_for_tuples::impl_for_tuples(30)]
+pub trait OnMessagesDelivered {
```

Comment:
Since the pallet is used on both source and target chains it's sometimes hard to identify which associated types are for which side.

A generic name like `OnMessagesDelivered` imho is equally valid as a callback for messages being delivered to the target chain and confirmations being delivered back to the source chain.

I'd vote for calling the handler (and the associated type) `OnDeliveryConfirmed`, cause I think it doesn't have this kind of ambiguity.
```suggestion
pub trait OnDeliveryConfirmed {
```